src/structures/VoiceChannelEffect.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const { Emoji } = require('./Emoji');
+
+/**
+ * Represents an effect used in a {@link VoiceChannel}.
+ */
+class VoiceChannelEffect {
+  constructor(data, guild) {
+    /**
+     * The guild where the effect was sent from.
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    /**
+     * The id of the channel the effect was sent in.
+     * @type {Snowflake}
+     */
+    this.channelId = data.channel_id;
+
+    /**
+     * The id of the user that sent the effect.
+     * @type {Snowflake}
+     */
+    this.userId = data.user_id;
+
+    /**
+     * The emoji of the effect.
+     * @type {?Emoji}
+     */
+    this.emoji = data.emoji ? new Emoji(guild.client, data.emoji) : null;
+
+    /**
+     * The animation type of the effect.
+     * @type {?AnimationType}
+     */
+    this.animationType = data.animation_type ?? null;
+
+    /**
+     * The animation id of the effect.
+     * @type {?number}
+     */
+    this.animationId = data.animation_id ?? null;
+  }
+
+  /**
+   * The channel the effect was sent in.
+   * @type {?VoiceChannel}
+   * @readonly
+   */
+  get channel() {
+    return this.guild.channels.cache.get(this.channelId) ?? null;
+  }
+
```

Comment:
```suggestion

/**
   * The user of this presence
   * @type {?User}
   * @readonly
   */
  get user() {
    return this.client.users.resolve(this.userId);
  }

```

Reply:
Just realize this doesn't extend Base, would that be possible?

Reply:
> Just realized this doesn't extend Base, would that be possible?

Idk structure was copied from jiralite pr lol

Reply:
About user getter i wasnt sure
as its a guild object
and as e.g. voice state it doesnt have user